src/Metadata/MetadataMatcher.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * This file is part of the prooph/event-store.
+ * (c) 2014-2016 prooph software GmbH <contact@prooph.de>
+ * (c) 2015-2016 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\EventStore\Metadata;
+
+use Prooph\EventStore\Exception\InvalidArgumentException;
+
+class MetadataMatcher
+{
+    private $data = [];
+
+    public function data(): array
+    {
+        return $this->data;
+    }
+
+    public function addMetadataMatch(string $key, Operator $operator, $value): void
```

Comment:
I'd prefer immutable objects for this kind of objects to avoid side effects. A new instance of the  class should be returned and the method should be named `withMetadataMatch`